hw/dv/tools/dvsim/sim.mk

```diff
@@ -97,9 +94,24 @@ ifneq (${sw_images},)
 			fi; \
 			echo "Building with command: $${bazel_cmd} build $${bazel_opts} $${bazel_label}"; \
 			$${bazel_cmd} build $${bazel_opts} $${bazel_label}; \
-			find -L $$($${bazel_cmd} info output_path)/ \
-				-type f -name "$${bazel_target}*" | \
-				xargs -I % sh -c 'cp -f % $${run_dir}/$$(basename %)'; \
+			for dep in $$($${bazel_cmd} cquery "labels(data, $${bazel_label})" \
+				--ui_event_filters=-info \
+				--noshow_progress \
+				--output=starlark); do \
+				if [[ $$dep != //hw* ]] && [[ $$dep != //util* ]] && [[ $$dep != //sw/host* ]]; then \
+					for artifact in $$($${bazel_cmd} cquery $${dep} \
+						--ui_event_filters=-info \
+						--noshow_progress \
+						--output=starlark \
+						--starlark:expr="\"\\n\".join([f.path for f in target.files.to_list()])"); do \
+						cp -f $${artifact} $${run_dir}/$$(basename $${artifact}); \
+						if [[ $$artifact == *.bin ]]; then \
+							cp -f "$${artifact%.bin}.elf" \
+								$${run_dir}/$$(basename "$${artifact%.bin}.elf"); \
+						fi; \
+					done; \
+				fi; \
+			done; \
```

Comment:
I think a comment explaining what is going on here would be helpful.

Reply:
done.